### PR TITLE
Dashboard to use queried data

### PR DESCRIPTION
**Pull request by mark-prins**
Adds some mock data for the dashboard and connects queries.
Ran into some trouble with the MSW batched mutation handler which wanted to grab my queries and I wasn't sure why!

Would still like to add in the stories and fix the layout, so will do another PR to finish off the issue.

### CODE REVIEW

File: packages/dashboard/src/api.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      isInbound: false,
+    });
+
+    if (result.invoiceCounts.__typename === 'InvoiceCountsConnector') {
+      return {
+        toBePicked: result.invoiceCounts.toBePicked ?? 0,
+      };
+    }
+
+    throw new Error('Argh! Outbound shipment statistics');
```
Comment (mark-prins) [inline packages/dashboard/src/api.ts:18]: oops. will update 🙄

File: packages/mock-server/src/data/database.ts [hunk 2]
```diff
@@ -7,6 +7,12 @@ import {
   NameData,
   removeElement,
 } from './data';
+import {
+  isAlmostExpired,
+  isExpired,
+  isThisWeek,
+  isToday,
+} from '@openmsupply-client/common/src/utils/dateFunctions';
```
Comment (mark-prins) [inline packages/mock-server/src/data/database.ts:15]: yeah, not ideal. Importing directly from `common` makes it choke - getting ENVIRONMENT without using node.. I think? can't remember.

File: packages/mock-server/src/worker/handlers.ts [hunk 3]
```diff
@@ -132,53 +132,75 @@ export const serverError = graphql.query('error500', (_, response, context) =>
...
+  (_, response, context) => {
+    const stockCounts = Api.ResolverService.statistics.stock();
 
-      return res;
-    });
+    return response(context.data({ stockCounts }));
   }
 );
 
+/**
```
Comment (mark-prins) [inline packages/mock-server/src/worker/handlers.ts:155]: yeah sorry. it was really annoying though!

**Review by joshxg — APPROVED** [review]

File: packages/common/src/schema.graphql [hunk 4]
```diff
@@ -612,6 +612,8 @@ type Queries {
     # sort options (only first sort input is evaluated for this endpoint)
     sort: [InvoiceSortInput!]
   ): InvoicesResponse!
+  invoiceCounts(isInbound: Boolean!): InvoiceCountsResponse!
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/common/src/schema.graphql:615]: yeah, once again I struggled with the name. Wanted to have a prefix for the aggregate queries and didn't come up with anything better than a 'count'

`InvoiceNodeType` is a much better idea. I really don't like the bools..  such as `isCustomer` when a type or enum is more flexible.

File: packages/dashboard/src/StatsPanel.tsx [hunk 5]
```diff
@@ -52,46 +58,57 @@ export const StatsPanel: FC<StatsPanelProps> = ({ stats, titleKey }) => {
...
-      <Grid container>
+      {isLoading ? (
         <Grid
-          item
-          display="flex"
-          justifyContent="flex-start"
+          style={{ height: 90, width: 300 }}
+          justifyContent="center"
           alignItems="center"
+          display="flex"
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/dashboard/src/StatsPanel.tsx:66]: went slightly further and pushed the loading indicator down, which helps with the layout too

**Review by mark-prins — COMMENTED** [review]
oops. I meant to comment but ended up reviewing the review of my PR 🙄

**Pull request merged**